### PR TITLE
fix: Mail client issue with port 465 on premise

### PR DESCRIPTION
**Pull request by ZhongpinWang**
Closes SAP/cloud-sdk-backlog#1250.

Relates to SAP/cloud-sdk-js#5124.

It seems like our last workaround in version `3.22.2` was causing the issue and it is also no longer needed. I tested it manually with the e2e test and mail send was successful both for 465 and 587.

I suggest that we merge this PR to have a canary release and check again if e2e pipeline works.

Reported to nodemailer -> https://github.com/nodemailer/nodemailer/issues/1684

### CODE REVIEW

**Review by deekshas8 — CHANGES_REQUESTED** [review]

File: packages/mail-client/src/mail-client.ts [hunk 1]
```diff
@@ -199,8 +155,11 @@ function createTransport(
     port: mailDestination.port
   };
 
-  if (mailDestination.proxyType === 'OnPremise' && socket) {
-    baseOptions.connection = socket;
+  if (mailDestination.proxyType === 'OnPremise') {
+    mailClientOptions = {
+      ...mailClientOptions,
```
[the remark this answers is not in the document]
Comment (ZhongpinWang) [reply packages/mail-client/src/mail-client.ts:160]: Thx, done

File: packages/mail-client/src/mail-client.ts [hunk 2]
```diff
@@ -140,56 +138,14 @@ async function createSocket(mailDestination: MailDestination): Promise<Socket> {
...
-  );
-}
-
 function createTransport(
   mailDestination: MailDestination,
-  mailClientOptions?: MailClientOptions,
-  socket?: Socket
+  mailClientOptions?: MailClientOptions
 ): Transporter<SentMessageInfo> {
-  const baseOptions: Options = {
```
[the remark this answers is not in the document]
Comment (ZhongpinWang) [reply packages/mail-client/src/mail-client.ts:192]: I am okay with leaving it here and importing from internal module. Added back.

File: packages/mail-client/src/mail-client.spec.ts [hunk 3]
```diff
@@ -239,98 +240,29 @@ describe('mail client', () => {
...
-      const { connection, createConnectionSpy } = mockSocketConnection();
       const spyCreateTransport = jest
         .spyOn(nodemailer, 'createTransport')
         .mockReturnValue(mockTransport as any);
       const spySendMail = jest
         .spyOn(mockTransport, 'sendMail')
-        .mockImplementation(() => {
-          connection.socket.on('data', () => {});
-        });
+        .mockImplementation(() => {});
```
[the remark this answers is not in the document]
Comment (ZhongpinWang) [reply packages/mail-client/src/mail-client.spec.ts]: Leftover, removed.

File: packages/mail-client/src/mail-client.ts [hunk 4]
```diff
@@ -199,8 +156,17 @@ function createTransport(
     port: mailDestination.port
   };
 
-  if (mailDestination.proxyType === 'OnPremise' && socket) {
-    baseOptions.connection = socket;
+  if (mailDestination.proxyType === 'OnPremise') {
+    if (!mailDestination.proxyConfiguration) {
```
Comment (deekshas8) [inline packages/mail-client/src/mail-client.ts:160]: [pp] Could you move this to a function so we don't have code duplication?

Comment (ZhongpinWang) [reply packages/mail-client/src/mail-client.ts]: I extracted the url build part out. Looks more clean now. Thanks for your suggestion.

**Pull request merged**